### ISSUE TITLE
Dataset query when training YOLO v3

### ISSUE THREAD

**Issue by golars497** [AlexeyAB/darknet#582]
Hi,

I am training YOLO v3 (352 by 352 as this is the highest resolution my GPU allows at the moment) to detect a single class. I am training to detect knives. The knife would be positioned around 2 - 3 meters from the camera (1280 by 720). Since YOLO v3 is a lot better than YOLO v2 when it comes to picking up small objects would it be better if I make the dataset so that the knives in the input images aren't small but maybe less than 1 meter from the camera? or shall I go about the default way and position them like I would when I am detecting the knives in real life (2-3 meters)

Many Thanks

**Comment by AlexeyAB** [comment]
Hi,

If images from training and detection datasets have the same resolution (1280 by 720), then objects should have the same size, i.e. knives should be positioned 2-3 metters from camera.

**Comment by golars497** [comment]
@AlexeyAB 

I wanted to ask something else as well,
So basically, because I have access to a number of computers I am training 5 different models:
YOLOv2 at 416 by 416px
YOLOv2 at 544 by 544 px
TINY-YOLOv2 at 448 by 448px
TINY-YOLOv2 at 544 by 544ps
YOLOv3 at 352 by 352 px

I am detecting 2 classes: knife and knives (when a bunch of knives are grouped). Actually I only want to detect 1 class but because during the construction of the data set I accidentally left a bunch of knives piled up somewhere in the input image so I decided I'd label because I was worried that if I didn't label it, it may hinder the models ability to detect. As I say in my first comments, the knives are place 2-3 meters away from camera.

For **positive samples** I have roughly **2200 self-made** images with knifes (most around 1280 by 720 px) and **800 close-up images** of knives from COCO website (most around 500px by 375px).

As for **negative samples**, I have **100 self-made** images (1280 by 720) and 3000 random images from 2007-2012 VOC (mostly 500px by 375)

I am using the same dataset for all of them.
Now, for what I would like to actually ask - I feel like the avg loss for Tiny-yolo 448 and 544 isn't getting any lower than 1.2 after many iterations and even at 20,000+ iterations I am getting some nans. I also, feel like the same might happen to YOLOv2 416 and 544. I  I wanted to know if this was expected?

**YOLOv2 at 416 by 416px**
![yolov2-416](https://user-images.githubusercontent.com/16651326/38430385-01881a5c-39b9-11e8-8576-db6431675afa.jpg)

**YOLOv2 at 544 by 544 px**
![yolov2-544px](https://user-images.githubusercontent.com/16651326/38430693-dd80d1c0-39b9-11e8-9fa5-1468fa3129e8.jpg)


**TINY-YOLOv2 at 448 by 448px**
![tiny-yolo-448](https://user-images.githubusercontent.com/16651326/38430609-a6809a02-39b9-11e8-9f48-cfd5313e5bf1.jpg)

**TINY-YOLOv2 at 544 by 544px**
![tiny-yolo-544px](https://user-images.githubusercontent.com/16651326/38430619-a9cda344-39b9-11e8-8b3c-e4ff47d33cca.jpg)

**YOLOv3 at 352 by 352 px**
![yolov3-352px](https://user-images.githubusercontent.com/16651326/38430625-ad54cba0-39b9-11e8-8a6e-738c16570ecc.jpg)

The cfg files:
[phase-2-training-analysis-cfg.zip](https://github.com/AlexeyAB/darknet/files/1884648/phase-2-training-analysis-cfg.zip)


I am currently using the darknet map function to get mAP against validation dataset similar data to self made images).

Sorry for pestering you and once again thank you for your help :)

**Comment by golars497** [comment]
Ah yes! Sorry I did mean 0.12 not 1.2. I've used darknet map command on all the weights generated for tiny-yolo-544 (with detection resolution set to 544) and the highest mAP I get is around 8-9%. I will send the full text file of results when I get on a pc

Many thanks

**Comment by golars497** [comment]
I did some more testing on TINY-YOLO-544. I took another video waving around a knife but in a well-lit area (compared to the one in my validation set which was kind of in the middle) and it works a lot better (I didnt calculate mAP as I didnt have time to put annotations on the video frames). Which makes me realize that I still need more diversity with my training data if I want to improve detection. I trained it on 544px but it works well even on 512px or 608px

**Comment by AlexeyAB** [comment]
Good training data + also if you want to train with more variety of lighting, you can train with 
`exposure = 2.5` or `exposure = 4.0` 
https://github.com/AlexeyAB/darknet/blob/1b1221ace5f49b5b464cf7f4ff89f31cf03d3da1/cfg/yolov3.cfg#L15
